release/src/release-notes.ts

```diff
@@ -118,97 +120,97 @@ enum ProductCategory {
   other = "Other",
 }
 
-const getIssueType = (issue: Issue): IssueType => {
-  if (isNonUserFacingIssue(issue)) return IssueType.underTheHoodIssues;
-  if (isAlreadyFixedIssue(issue)) return IssueType.alreadyFixedIssues;
-  if (isBugIssue(issue)) return IssueType.bugFixes;
-  return IssueType.enhancements;
+type CategoryIssueMap = Record<Partial<ProductCategory>, Issue[]>;
+
+type IssueMap = {
+  [IssueType.bugFixes]: CategoryIssueMap;
+  [IssueType.enhancements]: CategoryIssueMap;
+  [IssueType.alreadyFixedIssues]: CategoryIssueMap;
+  [IssueType.underTheHoodIssues]: CategoryIssueMap;
 };
 
-const addIssueToCategory = (
-  issueMap: Record<IssueType, Partial<Record<ProductCategory, Issue[]>>>,
-  issue: Issue,
-  issueType: IssueType,
-  productCategory: ProductCategory,
-) => {
-  if (!issueMap[issueType][productCategory]) {
-    issueMap[issueType][productCategory] = [];
-  }
-  issueMap[issueType][productCategory]!.push(issue);
+const getIssueType = (issue: Issue): IssueType => {
+  return match(issue)
+    .when(isNonUserFacingIssue, () => IssueType.underTheHoodIssues)
+    .when(isAlreadyFixedIssue, () => IssueType.alreadyFixedIssues)
+    .when(isBugIssue, () => IssueType.bugFixes)
+    .otherwise(() => IssueType.enhancements);
```

Comment:
this was my code, but it was a good opportunity to clean up some messy ifs into a nice clean [`ts-pattern`](https://github.com/gvergnaud/ts-pattern) call